### PR TITLE
Updated the javascript viewer to load vim 1.0

### PR DESCRIPTION
**Pull request by vim-sroberge**

### CODE REVIEW

**Review by mavimaec — APPROVED** [review]
Thanks @vim-sroberge !

**Comment by vim-sroberge** [comment]
# WebGL Order of Execution

## Milestone 0
- Strip the old references to Ara3D
  - Rename "araviewer" to "vimviewer"
- Hide the dat.gui controls on default
- Add VIM branding to the viewer – this is for the public Repo, and for a VIM hosted viewer, if EE wants to fork and remove branding they can.
- Create configuration options for EE ???
- Host the viewer on a high-performance CDN, so EE can use it (tbd - does EE host their fork?)
- Document best practices for hosting and serving VIMs
  - Compress them  
  - Put them on a server (e.g. azure blob) 
  - Keep the name ".vim" 
  - Serve it with the current content type (gzip) 
  - Let the client unencode 
- VimApi
  - instantiated/returned from current ara.view
  - Dispose()
  - ElementManager:
    - getElementIndices(): int[]
    - getBoundingBoxFromElement(int elementIndex)
    - getBoundingBoxFromElementCollection(int[] elementIndex)
    - setElementVisibility(int elementIndex, bool isVisible): void
      - disable mergeSingleInstances (to provide simplified implementation for toggling visibility of   individual elements)
      - (some complexity here to bind to ThreeJS InstancedMesh)
      - How do we modify the visibility of instances inside an InstancedMesh?
        - Juggle matrices and count (requires bookkeeping)
    - setElementCollectionVisibility(int[] elementIndices, bool isVisible): void
  - SceneManager:
    - getBoundingBox(): box
  - CameraManager:
    - Frame scene on load (i.e. position camera, and set its orientation)
    - frameToBoundingBox(box)

## Milestone 2
- Add a spinner for showing progress while loading the VIM 
  - (Percent done will be hard to compute, since served zip content, does not tell the total size, could parse the first chunk of the BFAST for that, and then guess) 
 - Need a callback in the "options" for the progress. 
- Integrate previous camera controls: see https://github.com/vimaec/vimaec.github.io/blob/master/js/CustomOrbitControls.js)
- (to possibly optimize) per-face vertex duplication (and color vertices)

## Milestone 3
- Add "BVH" builder as optional step controlled by options  
  - See old "WonderVIM" viewer
  - https://github.com/vimaec/vimaec.github.io/blob/master/js/three-mesh-bvh.js
- Retrieve element index from mouse position (screen coordinates), if BVH is built
- Api updates:
  - getElementIndex(screenCoordinates): integer

**Review by mavimaec — APPROVED** [review]
Looking good!

File: package.json [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
-  "bugs": "https://github.com/ara3d/ara3d-viewer/issues",
+  "main": "./dist/vim-webgl-viewer.js",
+  "homepage": "https://github.com/vimaec/vim-webgl-viewer.git",
+  "bugs": "https://github.com/vimaec/vim-webgl-viewer/issues",
   "license": "MIT",
   "author": {
-    "name": "Ara3D.com",
-    "email": "cdiggins@gmail.com"
+    "name": "VIM",
+    "email": "christopher.diggins@vimaec.com"
```
Comment (cdiggins) [inline package.json:11]: Please change this to "hello@vimaec.com"

**Review by cdiggins — APPROVED** [review]
Please leave the name "VIMLoader.js" for consistency with other Three.JS loaders (e.g. https://threejs.org/docs/#examples/en/loaders/GLTFLoader). In the future we may submit it to Three.JS as a type of loader. :-) 

Couple other minor changes requested, great work thanks!

**Pull request merged**